trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/plugins/generator/cacsl2boogietranslator/witness/YamlExtractedCorrectnessWitness.java

```diff
@@ -47,15 +47,15 @@ public class YamlExtractedCorrectnessWitness implements IExtractedCorrectnessWit
 	private final HashRelation<IASTNode, ExtractedWitnessInvariant> mInvariants = new HashRelation<>();
 	private final HashRelation<IASTNode, ExtractedFunctionContract> mFunctionContracts = new HashRelation<>();
 
-	public void addInvariant(final IASTNode node, final ExtractedWitnessInvariant entry) {
+	void addInvariant(final IASTNode node, final ExtractedWitnessInvariant entry) {
 		mInvariants.addPair(node, entry);
 	}
 
-	public void addInvariants(final Map<IASTNode, ? extends ExtractedWitnessInvariant> invariantMap) {
+	void addInvariants(final Map<IASTNode, ? extends ExtractedWitnessInvariant> invariantMap) {
 		invariantMap.forEach(this::addInvariant);
 	}
 
-	public void addFunctionContract(final IASTNode function, final ExtractedFunctionContract contract) {
+	void addFunctionContract(final IASTNode function, final ExtractedFunctionContract contract) {
```

Comment:
Do you omit the access modifier to want the methods to be package-private?
If you don't want to use them outside of the package in the future, then that's a good decision.

Reply:
Yes, these methods should be only accessible during construction by `YamlCorrectnessWitnessExtractor`, only the getters should be public. So maybe it would be even better to make this implementation a nested class in `YamlCorrectnessWitnessExtractor` (as it is also rather small and simple).

Reply:
Sure, you could do it as a nested class, but I think you can leave it as it is.